### PR TITLE
chore(ci): upgrade workflows

### PR DESCRIPTION
**Pull request by ivuorinen**
This pull request updates several GitHub Actions workflow files to improve concurrency handling, update action versions, and clarify permissions and job naming. The changes help ensure workflows do not run concurrently in conflicting ways, improve security, and make workflow files easier to understand and maintain.

**Workflow concurrency improvements:**

* Added `concurrency` groups to `.github/workflows/lint.yml`, `.github/workflows/pr-lint.yml`, and `.github/workflows/stale.yml` to prevent overlapping runs of the same workflow, reducing resource conflicts and redundant work. [[1]](diffhunk://#diff-107e910e9f2ebfb9a741fa10b2aa7100cc1fc4f5f3aca2dfe78b905cbd73c0d2R13-R16) [[2]](diffhunk://#diff-96c941d62860ad493e2ef6e2e641b5cdda1373729143a0a72a44f11f46895e4dR12-R35) [[3]](diffhunk://#diff-b5e0854ed0838024d5cc25b4e030922e34648b9ea8c432807e35495fb805b894R13-R24)

**Action and configuration updates:**

* Updated `actions/checkout` to version `v7.0.0` in `.github/workflows/lint.yml` and set `persist-credentials: false` for improved security.

**Workflow clarity and maintainability:**

* Added descriptive job names and expanded comments on permissions in `.github/workflows/pr-lint.yml` and `.github/workflows/stale.yml` for better readability and maintainability. [[1]](diffhunk://#diff-96c941d62860ad493e2ef6e2e641b5cdda1373729143a0a72a44f11f46895e4dR12-R35) [[2]](diffhunk://#diff-b5e0854ed0838024d5cc25b4e030922e34648b9ea8c432807e35495fb805b894R13-R24)

### CODE REVIEW

**Comment by ivuorinen** [comment]
@copilot resolve the merge conflicts in this pull request

File: docs/audit/findings/audit/open/NP-017.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+id: NP-017
+auditor: audit
+severity: advisory
+category: reliability
+area: functions/nvm_find_nvmrc.fish
+status: open
+found: 2026-05-08
```
[the remark this answers is not in the document]
Comment (ivuorinen) [reply docs/audit/findings/audit/open/NP-017.md]: Implemented.

Verified against the code before resolving, not just the record: `functions/nvm_find_nvmrc.fish:39` walks with `set dir (dirname "$dir")` — string manipulation, never a filesystem traversal — and line 36 breaks unconditionally at `if test "$dir" = /`. No symlink is ever followed, so a circular symlink cannot produce an infinite loop. The Problem section was disproven by its own Evidence.

Resolved as `invalid` through the findings-store workflow (`findings.py resolve NP-017 --status invalid`) rather than a hand edit, per `.claude/rules/audit-artifacts.md`, and `INDEX.md` was regenerated. The open file is gone; the record now lives in `resolved.jsonl` with the reasoning above.

**Comment by ivuorinen** [comment]
Re: the outside-diff note on `docs/audit/findings/ci/open/ci-c2676c57.md` 43-58 — replying here since that one had no inline thread to reply to.

Implemented, and the critique was correct: the append step lived only in a `#` comment, so copy-pasting the block would `PUT` the ruleset back unchanged and add no rule at all.

Replaced with an executable `jq` transform, dry-run verified against the live ruleset:

```
keys after strip:  ["bypass_actors","conditions","enforcement","name","rules","target"]
rule types:        [...,"required_status_checks"]
contexts:          ["Lint codebase","Lint on pull request"]
```

Two things beyond the original note:

- `PUT` also rejects the server-managed fields the `GET` returns (`id`, `node_id`, `_links`, `created_at`, `updated_at`, `source`, `source_type`, `current_user_can_bypass`), so the transform strips those as well. Without that the request fails even with a correct append.
- Verification now asserts the two contexts landed rather than that the rule type exists, as suggested — a `required_status_checks` rule with an empty array gates nothing.

Check-run names were confirmed against this PR's head via `gh api repos/.../commits/<sha>/check-runs` rather than inferred from the workflow YAML. `Lint on push` is deliberately excluded and documented as such: it is skipped on `pull_request` events (`if: github.event_name == 'push'`), so requiring it would block every merge.

File: docs/audit/findings/ci/open/ci-c2676c57.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            | .parameters.required_status_checks[].context'
+
+Expected output, exactly these two lines:
+
+    Lint codebase
+    Lint on pull request
+
+`Lint on push` is deliberately excluded: it is skipped on `pull_request`
+events (`if: github.event_name == 'push'`), and a required check that never
+reports blocks every merge.
```
[the remark this answers is not in the document]
Comment (ivuorinen) [reply docs/audit/findings/ci/open/ci-c2676c57.md]: Implemented — you're right, and the rationale was mine to fix.

Confirmed from this PR's own data rather than reasoning about it: the check-runs on the head commit list

```
Lint on push          [skipped]
Lint on pull request  [success]
Lint codebase         [success]
```

`Lint on push` reported a conclusion, so it was never pending. `.github/workflows/pr-lint.yml:20` puts the condition at job level (`if: github.event_name == 'push'`) while the workflow itself triggers on both `push` and `pull_request` — so on a PR the workflow always runs and both jobs always report.

The finding now states the correct reason: requiring `Lint on push` would not block anything, it would add a permanently green check that validates no pull-request linting at all. I also documented where the pending-check trap *does* apply — workflow-level skips (`paths`, `paths-ignore`, `branches`, `branches-ignore`) that stop the workflow from starting, so no check-run is ever created — so the distinction doesn't get lost the next time someone reads this.

One disclosure: the same wrong claim also survives in commit `a944ee0`'s message. I'm deliberately not fixing that — correcting a pushed commit message means another rebase and force-push, which would re-invalidate the review anchors on this PR for a prose error in history. The finding file is the authoritative artifact and is now correct.

File: docs/audit/findings/ci/open/ci-c2676c57.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+              "do_not_enforce_on_create": false,
+              "required_status_checks": [
+                {"context": "Lint codebase"},
+                {"context": "Lint on pull request"}
+              ]
+            }
+          }]' /tmp/rs.json > /tmp/rs-updated.json
+
+    gh api -X PUT repos/ivuorinen/nvm-auto-use.fish/rulesets/4676324 \
+      --input /tmp/rs-updated.json
```
[the remark this answers is not in the document]
Comment (ivuorinen) [reply docs/audit/findings/ci/open/ci-c2676c57.md]: Implemented, with one deliberate deviation.

Added a re-fetch-and-compare step before the `PUT`: the procedure now fetches the ruleset a second time, normalises both with `jq -S`, and diffs them. Any output means the ruleset changed since the original fetch, and the operator starts over instead of writing.

The finding now states explicitly that this **narrows** the read-modify-write window rather than closing it — GitHub documents no `If-Match` or conditional-update support on the rulesets endpoint, so a concurrent edit landing between the `diff` and the `PUT` would still be overwritten — and that the procedure should be run when nobody else is editing repository rulesets. That matches your point that the check cannot eliminate the race.

The deviation: I did not use `diff <(...) <(...)` or an `if/else` guard. This repository's shell is `fish`, where process substitution does not exist and `if` syntax differs from POSIX, and the block is a copy-paste procedure for a human operator. It is written as a plain command sequence — write both normalised files, run `diff`, and continue only if it is silent — so it pastes cleanly into `fish`, `bash`, and `sh` alike.

File: docs/audit/findings/ci/open/ci-c2676c57.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+            "type": "required_status_checks",
+            "parameters": {
+              "strict_required_status_checks_policy": true,
+              "do_not_enforce_on_create": false,
+              "required_status_checks": [
+                {"context": "Lint codebase"},
+                {"context": "Lint on pull request"}
+              ]
+            }
+          }]' /tmp/rs.json > /tmp/rs-updated.json
```
[the remark this answers is not in the document]
Comment (ivuorinen) [reply docs/audit/findings/ci/open/ci-c2676c57.md]: Implemented — correct catch, the append was not rerunnable.

`.rules += [...]` became:

```jq
.rules = ((.rules | map(select(.type != "required_status_checks"))) + [ { ... } ])
```

Filtering the type out before appending replaces any existing rule instead of stacking a second one.

Verified by feeding the transform's own output back through it, which is exactly the "second run" case:

```
pass1 rsc count:  1
pass2 rsc count:  1
pass1 == pass2 ?  IDENTICAL
contexts:         ["Lint codebase","Lint on pull request"]

old append form, run twice:
append-twice rsc count: 2
```

So the duplicate rules and repeated contexts you predicted did occur with the old form, and no longer do. Re-running the whole procedure is now a no-op when the rule is already correct.

**Comment by ivuorinen** [comment]
@coderabbitai review

File: functions/nvm_doctor.fish [hunk 5]
```diff
@@ -319,14 +315,14 @@ function _nvm_doctor_check_cache -d "Check cache status and health"
...
-        if test -n "$old_files"
-            echo "   ℹ️  Found "(count (string split '\n' "$old_files"))" cache files older than 7 days"
+        # `find` already returns one element per file; counting it directly.
+        # The previous `count (string split '\n' "$old_files")` always yielded 1:
+        # quoting joins the list with spaces, and '\n' in single quotes is a
+        # literal backslash-n that never matches.
+        set -l old_files (find "$cache_dir" -type f -mtime +7 2>/dev/null)
+        set -l old_count (count $old_files)
+        if test $old_count -gt 0
+            echo "   ℹ️  Found $old_count cache files older than 7 days"
```
[the remark this answers is not in the document]
Comment (ivuorinen) [reply functions/nvm_doctor.fish]: Implemented — confirmed against real files rather than the man page:

```
files:  144h=6d  168h=7d  180h=7.5d  192h=8d  216h=9d

find -mtime +6  ->  168h 180h 192h 216h     (7d and up)
find -mtime +7  ->            192h 216h     (8d and up)
```

So `+7` was silently skipping exactly the 7-to-8-day window the message claimed to cover. Switched to `-mtime +6`.

I also reworded the message from "older than 7 days" to "7 days or older", since `+6` matches files aged *at least* 7 days and the old phrasing is ambiguous at exactly 7.0 days.

One thing worth flagging: the regression test I added earlier could not have caught this. It dates its files to 2020, which matches either spelling. Added a boundary case that creates a file aged 7.5 days and asserts it is counted — mutation-verified, it passes on `+6` and fails on `+7`. `touch -d <relative>` is GNU-only, so that assertion is skipped with a notice on BSD/macOS instead of failing there.

**Comment by ivuorinen** [comment]
@coderabbitai review

**Comment by ivuorinen** [comment]
@coderabbitai review
